### PR TITLE
OpenAPI specifications for GET/POST source files - without attribute parameter

### PR DESCRIPTION
**Pull request by rubhanazeem**
Specifications for `GET/POST source/:project/:package/:binary/_attribute/`

### CODE REVIEW

File: src/api/public/apidocs-new/paths/source_project_name_package_name_binary_filename_attribute.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
+get:
+  summary: Get list of attributes and attribute values
```
Comment (saraycp) [inline src/api/public/apidocs-new/paths/source_project_name_package_name_binary_filename_attribute.yaml:2]: What about mentioning the binary/file here? This definition is maybe too generic :thinking:

File: src/api/public/apidocs-new/paths/source_project_name_package_name_binary_filename_attribute.yaml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      description: Bad request
+      content:
+        application/xml; charset=utf-8:
+          schema:
+            $ref: '../components/schemas/api_response.yaml'
+          examples:
+            invalid_attribute:
+              value:
+                code: invalid_attribute
+                summary: "Attribute '{attribute_name}' must be in the $NAMESPACE:$NAME style"
```
Comment (saraycp) [inline src/api/public/apidocs-new/paths/source_project_name_package_name_binary_filename_attribute.yaml:35]: This error can't happen in this case, as we don't pass the `attribute_name`.

File: src/api/public/apidocs-new/paths/source_project_name_package_name_binary_filename_attribute.yaml [hunk 3]
```diff
@@ -0,0 +1,72 @@
+get:
+  summary: Get list of attributes and attribute values
+  description: |
+    Get a list of all attributes and their values
```
Comment (saraycp) [inline src/api/public/apidocs-new/paths/source_project_name_package_name_binary_filename_attribute.yaml:4]: Same here, I'd mention the binary somehow.

**Review by saraycp — CHANGES_REQUESTED** [review]

**Review by saraycp — CHANGES_REQUESTED** [review]

File: src/api/public/apidocs-new/paths/source_project_name_package_name_binary_filename_attribute.yaml [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    description: Attributes you want to update
+    content:
+      application/xml; charset=utf-8:
+        schema:
+          $ref: '../components/schemas/source/attributes.yaml'
+        example:
+          attribute:
+            name: MaintenanceProject
+            namespace: OBS
+            binary: Ceph
```
[the remark this answers is not in the document]
Comment (saraycp) [reply src/api/public/apidocs-new/paths/source_project_name_package_name_binary_filename_attribute.yaml:63]: I know you added `binary` to the schema in the [other PR](https://github.com/openSUSE/open-build-service/pull/14009), but that's not the only problem here.

File: src/api/public/apidocs-new/paths/source_project_name_package_name_binary_filename_attribute.yaml [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        application/xml; charset=utf-8:
+          schema:
+            $ref: '../components/schemas/source/attributes.yaml'
+          example:
+            - name: AllowSubmitToMaintenanceRelease
+              namespace: OBS
+              binary: SLES-cd-DVD-x86_64
+            - name: OwnerRootProjectTest
+              namespace: OBS
+              binary: Ceph
```
Comment (saraycp) [inline src/api/public/apidocs-new/paths/source_project_name_package_name_binary_filename_attribute.yaml:24]: This is not correctly displayed by Swagger UI, but the reason is that you added the binary to the schema in the [previous PR](I know you added `binary` to the schema in the [other PR](https://github.com/openSUSE/open-build-service/pull/14009)) (which should be merged before this one).

Screenshot 2023-03-17 at 18-50-39 Swagger UI

**Review by saraycp — CHANGES_REQUESTED** [review]

File: src/api/public/apidocs-new/paths/source_project_name_package_name_binary_filename_attribute.yaml (hunk 1 above)
Comment (rubhanazeem) [reply src/api/public/apidocs-new/paths/source_project_name_package_name_binary_filename_attribute.yaml]: Done. It's updated

File: src/api/public/apidocs-new/paths/source_project_name_package_name_binary_filename_attribute.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (rubhanazeem) [reply src/api/public/apidocs-new/paths/source_project_name_package_name_binary_filename_attribute.yaml]: It's fixed now

**Pull request merged**